question10/main.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+// loading img 
+const wrapper = document.getElementById('wrapper');
+const loadImg = document.createElement('img');
+const showLoadImg = () => {
+    loadImg.src = "loading-circle.gif";
+    wrapper.appendChild(loadImg);
+} 
+
+const contentsList = [{to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"}, 
+                      {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}]
+
+
+const promise = new Promise((resolve) => {
+    setTimeout(() =>{
+      resolve(contentsList);
+    }, 3000)
+})
+
+async function asyncFunc() {
+    showLoadImg();
+    try {
+        const result = await promise;
+        createList(result);
+    }
+    catch(error) {
+        console.error(error);
+    }
```

Comment:
As it is now, I thought it would be better to show an error message if it cannot be retrieved, since nothing is shown on the screen. (This is just IMO)
IMO
```suggestion
    catch(error) {
        wrapper.textContent = "データが取得できませんでした"
        console.error(error);
    }
```

Reply:
Thank you for the review!! I really agree with you so I added the code you suggested.
Latest commit: https://github.com/csakyo/homework/pull/22/commits/cd20136eb81e7ad36cd08a75c7fbcd16c0e81805